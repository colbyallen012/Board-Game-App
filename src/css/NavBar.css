img {
  height: 25px;
  margin: 0 10px;
}

li {
  list-style: none;
  display: inline-block;
}

nav {
  display: flex;
  flex-flow: row;
  justify-content: space-around;
}
