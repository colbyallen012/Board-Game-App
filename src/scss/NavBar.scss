img {
  height: 25px;
  margin: 0 10px;
}

ul {
  display: inherit;
  padding: 0px;
}

li {
  display: inline-block;
  list-style: none;
}

nav {
  background-color: #444f68;
  display: flex;
  flex-flow: row;
  justify-content: space-around;
  align-content: center;
}

select {
  background-color: #DCE6EC;
  border-radius: 40px;
  box-shadow: 2px 2px #2eaebb;
  color: #444f68;
  height: 40px;
  margin: 40px 10px 25px 10px;
  width: 150px;
}

button {
  background-color: #DCE6EC;
  border-radius: 40px;
  box-shadow: 2px 2px #2eaebb;
  color: #444f68;
  height: 40px;
  margin: 25px 5px 25px 5px;
  width: 80px;
}

button.location {
  background-color: #DCE6EC;
  border-radius: 40px;
  box-shadow: 2px 2px #2eaebb;
  color: #444f68;
  height: 40px;
  margin: 40px 10px 25px 10px;
  width: 200px;
}

section {
  display: flex;
  flex-direction: row;
  justify-content: center;
  border-radius: 2px;
  margin:15px auto;
}

input {
  align-self: center;
  height: 100%;
  width: 200px;
  border-style: none;
  padding-left: 6px;
}

button.locationButton {
  background-color: #DCE6EC;
  border-radius: 40px;
  box-shadow: 2px 2px #2eaebb;
  color: #444f68;
  height: 40px;
  margin: 25px 10px 25px 10px;
  width: 125px;
}

 button.magnifyButton{
   border-top-left-radius: 40px;
   border-bottom-left-radius: 40px;
   border-top-right-radius: 0px;
   border-bottom-right-radius: 0px;
   margin:25px 0px 25px 0px;
   height: 42px; 
   width: 50px;
 }

input.searchLocation {
  border-top-right-radius: 40px;
  border-bottom-right-radius: 40px;
  height: 40px;
  box-shadow: 2px 2px #2eaebb;
}

button.changeGames {
  margin: 40px 10px 25px 0px;
  width: 90px;
}