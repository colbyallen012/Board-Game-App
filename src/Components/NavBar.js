import React, {Component} from 'react';
import App from '../App.js';
import '../scss/NavBar.scss'
import onePlayer from '../images/One-Player.svg'
import twoPlayers from '../images/Two-Players.svg'
import threePlayers from '../images/Three-Players.svg'
import fourPlayers from '../images/Four-Players.svg'
import fivePlayers from '../images/Five-Players.svg'
import groupPlayers from '../images/Group-Players.svg'
import search from '../images/Search.svg';

class NavBar extends Component {
  constructor(props) {
    super(props);
  }


  render() {
    console.log(this.props.status)
    if(this.props.status === "Locations"){
      return (
        <div>
      <nav>
        <ul>
          <li><button onClick={() => this.props.foodFilter()}>Has Food</button></li>
          <li><button onClick={() => this.props.drinkFilter()}>Has Drinks</button></li>
          <li><button onClick={() => this.props.sellerFilter()}>Sells Games</button></li>
          <li><button onClick={() => this.props.bringGameFilter()}>Bring Your Own Game</button></li>
        </ul>
        <section>
          <input type="text" placeholder="Search by Location" onChange={this.props.searchLowerCase}/>
          <button onClick={this.props.searchByText}><img src={search}/></button>
        </section>
        <button onClick={this.props.changeCards}>Games</button>
      </nav>
      </div>
      )
    }else{
      return (
        <div>
        <nav>
          <ul>
            <li><button onClick={() => this.props.playerFilter(1)}><img src={onePlayer} alt="One Player Games" /></button></li>
            <li><button onClick={() => this.props.playerFilter(2)}><img src={twoPlayers} alt="Two Player Games"/></button></li>
            <li><button onClick={() => this.props.playerFilter(3)}><img src={threePlayers} alt="Three Player Games" /></button></li>
            <li><button onClick={() => this.props.playerFilter(4)}><img src={fourPlayers} alt="Four Player Games" /></button></li>
            <li><button onClick={() => this.props.playerFilter(5)}><img src={fivePlayers} alt="Five Player Games" /></button></li>
            <li><button onClick={() => this.props.playerFilter(6)}><img src={groupPlayers} alt="Group Player Games" /></button></li>
          </ul>
          <select onChange={this.props.weightFilter}>
            <option>Weight </option>
            <option>Light</option>
            <option>Medium</option>
            <option>Heavy</option>
          </select>
          <select onChange={this.props.gameTypeFilter}>
            <option>Type</option>
            <option>Party</option>
            <option>Strategy</option>
            <option>Thematic</option>
            <option>Children's</option>
            <option>Family</option>
            <option>Abstract</option>
          </select>
          <button id="location" onClick={this.props.changeCards}>Locations</button>
        </nav>
        </div>
    )

    }
  }
}

export default NavBar;
