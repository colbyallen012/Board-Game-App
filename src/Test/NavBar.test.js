import React from 'react';
import ReactDOM from 'react-dom';
import NavBar from '../Components/NavBar';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const mockView = "Games"

describe('NavBar', () => {
  let wrapper;
  
  beforeEach(() =>{
    wrapper = shallow(
      <NavBar
      // games={this.props.gamesData}
      // changeCards = {this.changeCards}
      // playerFilter={this.props.playerFilter}
      // weightFilter={this.props.weightFilter}
      // gameTypeFilter={this.props.gameTypeFilter}
      // oodFilter={ this.props.foodFilter }
      // drinkFilter={this.props.drinkFilter }
      // sellerFilter={ this.props.sellerFilter }
      // bringGameFilter={ this.props.bringGameFilter } 
      />
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
   it('should toggle between Games and Locations', () => {
    wrapper.find('.toggle-button').simulate('click', { target: { value: 'Games' } });
    expect(wrapper.state('status')).toEqual("Locations");

  });
});