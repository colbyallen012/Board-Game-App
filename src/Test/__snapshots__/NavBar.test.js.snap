// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match the snapshot with all data passed in correctly 1`] = `
<div>
  <nav>
    <ul>
      <li>
        <button
          onClick={[Function]}
        >
          <img
            alt="One Player Games"
            src="One-Player.svg"
          />
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          <img
            alt="Two Player Games"
            src="Two-Players.svg"
          />
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          <img
            alt="Three Player Games"
            src="Three-Players.svg"
          />
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          <img
            alt="Four Player Games"
            src="Four-Players.svg"
          />
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          <img
            alt="Five Player Games"
            src="Five-Players.svg"
          />
        </button>
      </li>
      <li>
        <button
          onClick={[Function]}
        >
          <img
            alt="Group Player Games"
            src="Group-Players.svg"
          />
        </button>
      </li>
    </ul>
    <select
      onChange={[MockFunction]}
    >
      <option>
        Difficulty
      </option>
      <option>
        Light
      </option>
      <option>
        Medium
      </option>
      <option>
        Heavy
      </option>
    </select>
    <select
      onChange={[MockFunction]}
    >
      <option>
        Type
      </option>
      <option>
        Party
      </option>
      <option>
        Strategy
      </option>
      <option>
        Thematic
      </option>
      <option>
        Children's
      </option>
      <option>
        Family
      </option>
      <option>
        Abstract
      </option>
    </select>
    <button
      className="location toggle-button"
      id="location"
      onClick={[MockFunction]}
    >
      Locations
    </button>
  </nav>
</div>
`;
